backslash/session.py

```diff
@@ -55,6 +55,11 @@ def report_test_start(self, name, file_name=NOTHING, class_name=NOTHING, test_lo
 
         return returned
 
+    def report_upcoming_tests(self, tests):
+        if 'append_upcoming_tests' in self.client.api.info().endpoints:
```

Comment:
This check doesn't belong here. If the server doesn't support our API and it was forced to report it, we should fail accordingly...